mesonbuild/dependencies/base.py

```diff
@@ -2061,6 +2061,9 @@ def __call__(self, env: Environment, for_machine: MachineChoice,
 
 def get_dep_identifier(name, kwargs) -> T.Tuple:
     identifier = (name, )
+    from ..interpreter import permitted_dependency_kwargs
+    assert len(permitted_dependency_kwargs) == 19, \
```

Comment:
I'm not entirely sure this is useful, since it wouldn't catch something like `embed`, which is probably pretty important to differentiate.

Reply:
It's just to be catched when someone adds new kwargs. Maybe just the comment in func_dependency() is enough?

Reply:
I don't know? I guess it's not hurting anything, I'm just worried that it gives a false sense of security